### PR TITLE
MAINT: Fixups to new functions in np.lib.recfunctions

### PR DESCRIPTION
**Pull request by ahaldane**
Followup to #11526 fixing up a few issues with that PR.

### CODE REVIEW

File: numpy/lib/recfunctions.py [hunk 1]
```diff
@@ -1164,12 +1166,13 @@ def require_fields(array, required_dtype):
...
-    This function assigns to from the old to the new array by name, so the
+    This function assigns from the old to the new array by name, so the
     value of a field in the output array is the value of the field with the
-    same name in the source array.
+    same name in the source array. This has the effect of creating a new
+    ndarray containing only the fields "required" by the required_dtype.
 
     If a field name in the required_dtype does not exist in the
-    input array, that field is set to 0 in the output array.
+    input array, that field is created and set to 0 in the output array.
```
Comment (eric-wieser) [inline numpy/lib/recfunctions.py]: Can you add an example of this too?

Comment (ahaldane) [reply numpy/lib/recfunctions.py]: done

File: numpy/lib/recfunctions.py [hunk 2]
```diff
@@ -1081,7 +1081,9 @@ def apply_along_fields(func, arr):
     Apply function 'func' as a reduction across fields of a structured array.
 
     This is similar to `apply_along_axis`, but treats the fields of a
-    structured array as an extra axis.
+    structured array as an extra axis. The fields are all first cast to a
+    common type following the type-promotion rules from `np.result_type`
```
Comment (eric-wieser) [inline numpy/lib/recfunctions.py]: I don't think this will link correctly - you need to use `numpy.result_type` or just `result_type`

**Review by eric-wieser — APPROVED** [review]

**Comment by eric-wieser** [comment]
CI fails with:

AssertionError: ignore filter should not be used; found in /numpy/build/testenv/lib/python3.6/site-packages/numpy/lib/recfunctions.py on line 970

**Comment by ahaldane** [comment]
Ignore-filter issue fixed.

**Comment by charris** [comment]
Thanks Allan.

**Pull request merged**